### PR TITLE
Since infinite traitor items is okay, I improved it

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
tweak: NULL Crate in cargo was added again because people are fucking idiots so if we're going to have this stupid fucking crate might as well make it not fucking garbo
/:cl:

### CODE REVIEW

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -147,11 +147,27 @@
 	name = "NULL_ENTRY"
 	hidden = TRUE
 	cost = 14000
-	contains = list(/obj/item/weapon/storage/box/syndicate)
+	contains = list()
```
Comment (optimumtact) [inline code/modules/cargo/packs.dm]: this should just be null to save a list init surely?

Comment (Iamgoofball) [inline code/modules/cargo/packs.dm]: okay, my concern is runtimes though

**Comment by Firecage** [comment]
Goof. Thankyou.

**Comment by Shadowlight213** [comment]
Wait....isn't this going to include nuke op items too, since the list isn't filtered by gamemode?
Absolute madman.

**Comment by Iamgoofball** [comment]
Surplus.

**Comment by KorPhaeron** [comment]
>previously a 20-30tc box
>now a 50tc box

why

**Comment by Iamgoofball** [comment]
We can always tweak it if it proves too powerful

**Comment by ma44** [comment]
@KorPhaeron it doesn't contain anything great, it doesn't have the old items like eswords, crossbows, implants, c4s and a whole lot of other nice stuff

Also below are all possible items able to be put into this crate, the crate also automatically filters out items if it's not nuke ops 

/datum/uplink_item/dangerous/revolver
/datum/uplink_item/ammo
/datum/uplink_item/stealthy_weapons/foampistol (comes with riot ammunition)
/datum/uplink_item/stealthy_weapons/soap
/datum/uplink_item/stealthy_weapons/traitor_chem_bottle
/datum/uplink_item/stealthy_weapons/dart_pistol (basically a syringe gun, pair with chem for results)
/datum/uplink_item/stealthy_weapons/suppressor (any small firearms)
/datum/uplink_item/stealthy_weapons/pizza_bomb
/datum/uplink_item/stealthy_weapons/door_charge
/datum/uplink_item/stealthy_tools/camera_bug
/datum/uplink_item/stealthy_tools/smugglersatchel
/datum/uplink_item/stealthy_tools/stimpack
/datum/uplink_item/stealthy_tools/mulligan
/datum/uplink_item/stealthy_tools/emplight
/datum/uplink_item/stealthy_tools/cutouts
/datum/uplink_item/suits (regular suit)
/datum/uplink_item/device_tools/binary (listen to AI)
/datum/uplink_item/device_tools/encryptionkey (Listen to all channels + communicate with traitors)
/datum/uplink_item/badass/syndiecards (robust playing cards!)

**Comment by BlueNothing** [comment]
So... what got taken out?

**Comment by Iamgoofball** [comment]
That's not the full list and you know it, don't mislead people like the slimeball you are.

**Comment by Jalleo** [comment]
Okay I can confirm by testing this shit it dosent give anything that is tied to specific gamemodes.
https://github.com/tgstation/tgstation/blob/master/code/modules/uplink/uplink_item.dm
It looks for everything outside of include modes but it does include stuff that is excluded from certain modes. e.g. unless its setup as exclude_modes = list().

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -147,11 +147,27 @@
 	name = "NULL_ENTRY"
 	hidden = TRUE
 	cost = 14000
-	contains = list(/obj/item/weapon/storage/box/syndicate)
+	contains = null
```
Comment (AnturK) [inline code/modules/cargo/packs.dm]: Iirc the cargo uses empty contains to differentiate between abstract and normal crates. So you need to actually put something in here or change that.

Comment (Iamgoofball) [inline code/modules/cargo/packs.dm]: Fuck

File: code/modules/cargo/packs.dm [hunk 3]
```diff
@@ -147,11 +147,27 @@
...
 	crate_name = "emergency crate"
 	crate_type = /obj/structure/closet/crate/internals
 	dangerous = TRUE
 
+/datum/supply_pack/emergency/syndicate/fill(obj/structure/closet/crate/C)
+	var/list/uplink_items = get_uplink_items()
```
Comment (Jalleo) [inline code/modules/cargo/packs.dm]: See my comment above for anyone unsure of what will actually be included currently.

File: code/modules/cargo/packs.dm (hunk 2 above)
Comment (Jalleo) [inline code/modules/cargo/packs.dm]: Just put in a dummy object to remove once you put the first syndicate gear in it.

**Comment by evsey9** [comment]
Can you still get uplink implants from this?

**Comment by evsey9** [comment]
I don't get why this is better. Can someone explain?

**Comment by Iamgoofball** [comment]
It makes the null crate work like a syndicate surplus crate

File: code/modules/cargo/packs.dm (hunk 3 above)
Comment (RemieRichards) [inline code/modules/cargo/packs.dm]: this list should be:
```var/static/list/uplink_items = get_uplink_items()```
(yes ```/static``` works inside of procs)

Comment (Iamgoofball) [inline code/modules/cargo/packs.dm]: ok

**Comment by ShadowDeath6** [comment]
winning the lottery with 50 card stacks when

**Comment by optimumtact** [comment]
Finish it already

**Comment by evsey9** [comment]
Can you get an uplink implant from the surplus crate?

**Comment by YakumoChen** [comment]
goofball seems rather salty that the cargo we had for years, that nobody complained about, is now unbalanced because of changes he himself made, and people don't want him to remove the cargo perks that have existed for years to justify his unbalancing.

**Comment by optimumtact** [comment]
would you fix it already please

**Comment by Iamgoofball** [comment]
done @optimumtact

**Comment by Bawhoppen** [comment]
what a meme :+1:

File: code/modules/cargo/packs.dm (hunk 1 above)
Comment (Jalleo) [inline code/modules/cargo/packs.dm]: It was stated by someone else that it cannot be null.

**Comment by evsey9** [comment]
if you can't get implants like uplink implants and microbomb implants from this then i don't want it

**Comment by evsey9** [comment]
Noooo
Why
I can't get implant crates anymore

**Comment by Iamgoofball** [comment]
@evsey9 implants can spawn from surplus crates

**Comment by evsey9** [comment]
Yeah i forgot you can buy uplink implants
Its just me being dumb
Ok this is actually good

**Comment by evsey9** [comment]
Do you even test your PRs, goof.

**Pull request merged**